client/src/proxmark3.c

```diff
@@ -996,6 +997,22 @@ int main(int argc, char *argv[]) {
             continue;
         }
 
+        if (strcmp(argv[i], "--ncpu") == 0) {
+            if (i + 1 == argc) {
+                PrintAndLogEx(ERR, _RED_("ERROR:") " missing CPU number specification after --ncpu\n");
+                show_help(false, exec_name);
+                return 1;
+            }
+            long int ncpus = strtol(argv[i + 1], NULL, 10);
+            if (ncpus < 0 || ncpus >= INT_MAX) {
```

Comment:
How about for the upper limit use the number of CPU reported by the system.

Reply:
I considered this but it felt like a bit of an artificial limitation. This is kind of advanced setting, so if one wants to mess with it for whatever reason, I felt I should not get in their way :)

Reply:
I see no reason why we would allow anyone to set invalid number.   It will only crash the software.  
Change and it can be merged.

Reply:
It should not crash anything — it will spawn more threads than physical cores, which will probably starve other processes of CPU time but shouldn't crash anything.

Anyways, I will cap this number and submit a change to this PR shortly.